Перенёс задания А и Б в свои функции, добавил тесты программы при запуске main функции

goutosama opened the pull request.

goutosama committed:
Переменные А и Б теперь в main, задания А и Б разделены по функциям

goutosama committed:
Merge branch 'Chadov_Sergej_Vadimovich' of https://github.com/fagirton/Informatics_2022 into Chadov_Sergej_Vadimovich

goutosama committed:
Merge branch 'Chadov_Sergej_Vadimovich' of https://github.com/fagirton/Informatics_2022 into Chadov_Sergej_Vadimovich

goutosama committed:
Попытка в импровизированные тесты заданий А и Б

goutosama committed:
Решение заданий с Codewars, также убрал повторяющиеся выводы в консоль

goutosama committed:
Перенес функции из main.go

jskonst commented on golang/test.go:
```diff
@@ -0,0 +1,30 @@
+package main
```
1 - давайте тесты перенесем в пакет с тестами

jskonst commented on golang/tasks.go:
```diff
@@ -0,0 +1,26 @@
+	}
+	return res
+}
+func taskB(a float64, b float64) []float64 {
```
x - тоже массив

jskonst commented on golang/codewars.go:
```diff
@@ -0,0 +1,39 @@
+}
+
+func countingSheep(x []bool) int {
+	var r int
```
не используйте однобуквенные переменные

jskonst commented on golang/test.go:
```diff
@@ -0,0 +1,30 @@
+
+import "fmt"
+
+func test() {
```
1 0 переносите в пакет тестами

jskonst commented on golang/test.go:
```diff
@@ -0,0 +1,30 @@
+import "fmt"
+
+func test() {
+	var resultsA = [6]float64{-2867.2049337558983, -879771.242424423, 60553.396028559044, 7705.642156268663, 2947.1559153687504, 1627.2100029870098}
+	var resultsB = [5]float64{-879771.242424423, 392039.14018305, 43864.686778786934, 6807.290024311177, 2651.777897551827}
```
не нужна такая точность - нам хватит до 3 знака - в тестах есть такая возможность указать точность сравнения https://github.com/stretchr/testify

jskonst commented on golang/test.go:
```diff
@@ -0,0 +1,30 @@
+	var resultsA = [6]float64{-2867.2049337558983, -879771.242424423, 60553.396028559044, 7705.642156268663, 2947.1559153687504, 1627.2100029870098}
+	var resultsB = [5]float64{-879771.242424423, 392039.14018305, 43864.686778786934, 6807.290024311177, 2651.777897551827}
+
+	const a float64 = 4.1
```
1 проверка 1 функция теста

jskonst commented on golang/test.go:
```diff
@@ -0,0 +1,30 @@
+	var resultsB = [5]float64{-879771.242424423, 392039.14018305, 43864.686778786934, 6807.290024311177, 2651.777897551827}
+
+	const a float64 = 4.1
+	const b float64 = 2.7
```
1 - нужно проверить только функцию вычислений
2 - в задаче а и б - можно тогда проверить только размеры получающихся массивов

goutosama committed:
Неужели нормальные тесты? + Нерабочая реализация Game of Life
Я так и не придумал хорошего способа избежать проверки соседей за пределами поля

goutosama committed:
Пока без gameOfLife потому что cognitive complexity 57 ((

goutosama committed:
Снова исправления, чтобы порадовать линтер

goutosama committed:
Разделение тестов по заданиям

goutosama committed:
Проблемы с параллельностью?

jskonst commented on golang/internal/tasks.go:
```diff
@@ -0,0 +1,26 @@
+
+import "math"
+
+func formula(x float64, a float64, b float64) float64 {
```
я б ее тоже сделал public (Formula) чтобы ее можно было проверить в тестах ну и логично что ее потенциально могут переисполоьзовать

jskonst commented on golang/internal/tasks.go:
```diff
@@ -0,0 +1,26 @@
+	return y
+}
+
+func TaskA(a float64, b float64) []float64 {
```
Диапазон изменения x передайте в качестве параметров функции

goutosama replied:
Только диапазон изменения? Начальное и конечное значения оставить внутри функции?

jskonst replied:
Диапазон подразумевает начальное и конечное значение :)

jskonst commented on golang/internal/tasks.go:
```diff
@@ -0,0 +1,26 @@
+func TaskA(a float64, b float64) []float64 {
+	var i float64 = 1.5
+	var res []float64
+	for ; i < 3.5; i = i + 0.4 {
```
```suggestion
	for i := xn ; i < xk; i += dx {
```

jskonst commented on golang/tests/test.go:
```diff
@@ -0,0 +1,47 @@
+func TestTaskA(t *testing.T) {
+	t.Parallel()
+
+	var resultsA = [6]float64{-2867.2049337558983, -879771.242424423, 60553.396028559044, 7705.642156268663, 2947.1559153687504, 1627.2100029870098}
```
1 - не надо с такой точностью - достаточно до 3 знаков - в проверках это можно указать
2 - Добавьте просто тест на функцию - вычисление 1 точки
3 - соответственно тест задачи А и Б - сокращается - только нужно проверить размер массива (а значения в тестах просто для функции 1 точки)
4 - добавьте проверку случаев когда напрмиер xk < xn

jskonst commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+golang/.vscode/settings.json
+golang/пускай подумает о своем поведении/
```
Не называйте файлы русскими буквами

goutosama replied:
Да, оставил эту папку для себя и не хотел, чтобы она вообще индексировалась гитом, но пришлось добавить сюда, чтобы не мешалась

goutosama committed:
Тесты, тесты и снова доделываем тесты

goutosama committed:
Линтер, снова

goutosama committed:
Гитигнор не хотел игнорить настройки VSC

goutosama committed:
Добавил решения последних двух задач с codewars

goutosama committed:
Merge branch 'Chadov_Sergej_Vadimovich' of https://github.com/fagirton/Informatics_2022 into Chadov_Sergej_Vadimovich

goutosama committed:
А вот и первое задание с codewars с польской коровой

goutosama committed:
- Ненужные комментарии

jskonst commented on golang/internal/codewars.go:
```diff
@@ -0,0 +1,87 @@
+//Codewars 2: Возвращение
+
+func PolishCow(text string) string {
+	var symbols = [][]string{
```
🔕 используйте словарь

jskonst commented on golang/internal/codewars.go:
```diff
@@ -0,0 +1,87 @@
+	return text
+}
+
+func Find_all(arr []string, search string) []int {
```
CamelCase

jskonst commented on golang/internal/tasks.go:
```diff
@@ -0,0 +1,25 @@
+	var x []float64 = []float64{1.9, 2.15, 2.34, 2.74, 3.16}
+	var res []float64
+	var i int8
+	for ; i < 5; i++ {
```
```suggestion
	for var i int8=0 ; i < len(x); i++ {
```

jskonst commented on golang/internal/tasks.go:
```diff
@@ -0,0 +1,25 @@
+	return res
+}
+func TaskB(a float64, b float64) []float64 {
+	var x []float64 = []float64{1.9, 2.15, 2.34, 2.74, 3.16}
```
Перенести в параметр функции

jskonst commented on golang/main.go:
```diff
@@ -2,26 +2,46 @@ package main
+	fmt.Println(internal.PaperworkCount(23, 2))
+	fmt.Println(internal.HeroWithGunShootsDragons(37, 20))
+
+	//internal.GameOfLife(1, 5, 5)
```
не оставляйте закомментированный код

jskonst commented on golang/tests/test.go:
```diff
@@ -0,0 +1,67 @@
+	const xn float64 = 1.5
+	const xk float64 = 3.5
+	const xd float64 = 0.4
+
```
Добавьте отдельные тесты на формулу
В этих тестах проверьте возможные значения передаваемые в формулу, особенно критические

jskonst commented on golang/tests/test.go:
```diff
@@ -0,0 +1,67 @@
+
+	fmt.Println("Task A: Test inputs")
+	if (xn < xk && xd > 0) || (xn > xk && xd < 0) {
+		fmt.Print("- Passed")
```
тест должен падать

jskonst replied:
В задаче а проверяем что возвращается срез ожидаемого размера
1 - нормальное вычисление 
2 xk > xn (0 значений)
3 dx > xk-xn (1 значение)

jskonst commented on golang/tests/test.go:
```diff
@@ -0,0 +1,67 @@
+}
+
+func TestTaskB(t *testing.T) {
+	t.Parallel()
```
1 - дассив длины >0
2 - пустой массив

goutosama replied:
Немного не понял, тест должен проверять какой длинны входной массив? Или выходной массив?
И какой смысл проверять на пустоту массива, если первым уже проверили, что его длина больше нуля?

jskonst replied:
Выходной

goutosama committed:
Исправил, но ещё не всё (есть код в комментах)

goutosama committed:
Штуки с тестами

goutosama committed:
Game Of Life (пока не готов - с проблемами)

goutosama committed:
Linter moment

goutosama committed:
Linter moment 2

jskonst commented on golang/internal/gameOfLife_true.go:
```diff
@@ -0,0 +1,76 @@
+package internal
```
На будущее - это бы в отдельный PR - только этот файл - потому что сейчас я не могу другие задачи завести не проверив эту :(

jskonst commented on golang/internal/tasks.go:
```diff
@@ -0,0 +1,23 @@
+package internal
```
Принято - все ок

jskonst commented on golang/tests/test.go:
```diff
@@ -0,0 +1,66 @@
+	var testResA []float64 = internal.TaskA(a, b, xn, xk, xd)
+
+	t.Run("Check 1 value", func(t *testing.T) {
+		t.Parallel()
```
а это вам зачем? в целом можно - но можно и не делать

goutosama replied:
Я немного не понял. А вы про что?)

jskonst commented on golang/tests/test.go:
```diff
@@ -0,0 +1,66 @@
+	const a float64 = 4.1
+	const b float64 = 2.7
+	// Не уверен, что строка ниже вообще будет работать, но пусть будет
+	assert.Nil(t, internal.Formula(1, a, b))
```
вы правы - не надо это

goutosama committed:
СКОБКИ. ВСЕМУ ВИНОЙ БЫЛИ СКОБКИ.

goutosama committed:
Опять не понял почему не работает
Похоже на зеркальное отражение сетки с поворотом

goutosama committed:
GAME OF LIFE РАБОТАЕТ??? Не окончательная версия
Скорее всего работает только с квадратными сетками

goutosama committed:
Линтер не хочет признавать мою победу

goutosama committed:
Простая внутренняя проверка сетки для Game of Life

goutosama committed:
Небольшие исправления в GoL

goutosama committed:
ОНО ПРАВДА РАБОТАЕТ, КАК НАДО (осталось переделать функцию проверки сетки)

goutosama committed:
Небольшой Clean-up в main, так и не сделал проверку сетки в GoL

goutosama committed:
Чайник с функцией Dead Inside
